public/app/plugins/datasource/cloudwatch/components/PanelQueryEditor.tsx

```diff
@@ -9,35 +9,28 @@ import LogsQueryEditor from './LogsQueryEditor';
 
 export type Props = ExploreQueryFieldProps<CloudWatchDatasource, CloudWatchQuery>;
 
-interface State {
-  queryMode: ExploreMode;
-}
-
-export class PanelQueryEditor extends PureComponent<Props, State> {
-  state: State = { queryMode: (this.props.query.queryMode as ExploreMode) ?? ExploreMode.Metrics };
-
-  onQueryModeChange(mode: ExploreMode) {
-    this.setState({
-      queryMode: mode,
-    });
-  }
+const apiModes = {
+  Metrics: { label: 'CloudWatch Metrics', value: 'Metrics' },
+  Logs: { label: 'CloudWatch Logs', value: 'Logs' },
+};
 
+export class PanelQueryEditor extends PureComponent<Props> {
   render() {
-    const { queryMode } = this.state;
+    const { query } = this.props;
+    const apiMode = query.apiMode ?? query.queryMode ?? 'Metrics';
```

Comment:
Removed the state and used the code from deleted component to get it from query. Please check if this has any side effect.